김현수/mvvmstudy/app/src/main/java/com/example/mvvmstudy/ViewModel/MainViewModel.kt

```diff
@@ -0,0 +1,47 @@
+package com.example.mvvmstudy.ViewModel
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import com.example.mvvmstudy.data.MainRepository
+import com.example.mvvmstudy.data.Repository
+
+class MainViewModel : ViewModel() {
+
+    private val mainRepository = MainRepository()
+
+    private var page: Int = 1
+
+    private var search: String = ""
+
+    private val _isLoading: MutableLiveData<Boolean> = MutableLiveData()
+    val isLoading: LiveData<Boolean>
+        get() = _isLoading
+
+    fun setCurSearch(q: String) {
+        search = q
+    }
+
+    fun getCurSearch(): String {
+        return search
+    }
+
+    fun getCurPage(): Int =
+        page++
+
+    fun initSearch() {
+        page = 1
+        mainRepository.reposClear()
+    }
+
+    fun searchRepositories(query: String, page: String) {
+        _isLoading.value = true
+        mainRepository.searchRepositories(query, page)
```

Comment:
isLoading false로 주는 부분이 빠진 것 같네요!
```suggestion
        _isLoading.value = true
        mainRepository.searchRepositories(query, page)
          _isLoading.value = false
```

Reply:
로딩은 구현하다가 말아가지고 실제론 안쓰는 부분이네욥 다시 수정해보겠습니다!